src/NUnitFramework/tests/Interfaces/TestMessageTests.cs

```diff
@@ -0,0 +1,61 @@
+// ***********************************************************************
+// Copyright (c) 2017 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using NUnit.Framework.Interfaces;
+using System;
+using System.Text;
+
+namespace NUnit.Framework.Tests.Interfaces
+{
+    [TestFixture]
+    class TestMessageTests
+    {
+        [TestCase("destination", "text", "testId")]
+        [TestCase("destination", "text", null)]
+        [TestCase(null, "text", "testId")]
+        public void ToXml_IncludeAttributesInProperFormatting(string destination, string message, string testId)
+        {
+            var testMessage = new TestMessage(destination, message, testId);
+            var expected = new StringBuilder("<test-message");
+
+            if (destination != null)
```

Comment:
If we allow the destination to be null, how will any extensions decide whether the message is for them? I know the original spec talks about broadcast, but that doesn't make much sense to me unless we require all extensions to respond to broadcasts, which would be a breaking change.

If you have a particular usage in mind, let us know. Otherwise, I think we should require a destination.

[Good thing about tests: I didn't notice this originally. 😄]

@jnm2 Opinion on this?

@nvborisenko I'm OK with merging this once we resolve the issue about a required destination. FYI, it takes two of the team members agreeing in order to merge, so someone else will need to add their voice.

Reply:
OK. It's ready as far as I'm concerned. We'll wait for the second review.

Reply:
BTW travis build returns unpredictable results. And I guess somebody is working on setting up AzureDevops pipelines.